jbmc/src/java_bytecode/java_types.cpp

```diff
@@ -839,6 +839,34 @@ bool equal_java_types(const typet &type1, const typet &type2)
   return (type1 == type2 && arrays_with_same_element_type);
 }
 
+std::vector<java_generic_parametert>get_all_generic_parameters(
+  const typet &type)
+{
+  PRECONDITION(
```

Comment:
:bulb: Now this is a more general function, perhaps returning an empty vector for_non_ generic types makes more sense than crashing (it isn't obvious from the name or signature that this only works on specific kinds of types).